osu.Game/Screens/Edit/Compose/Components/DistanceSnapGrid.cs

```diff
@@ -18,7 +18,7 @@ namespace osu.Game.Screens.Edit.Compose.Components
     /// <summary>
     /// A grid which takes user input and returns a quantized ("snapped") position and time.
     /// </summary>
-    public abstract class DistanceSnapGrid : CompositeDrawable
+    public abstract class DistanceSnapGrid : CompositeDrawable, IDistanceSnapProvider
```

Comment:
Is the interface spec here just because-we-can? Or is it used in some way I can't see?

Reply:
It's more about that `DistanceSnapGrid` now provides its own snap provider logic which builds over the `HitObjectComposer` one, but with the distance spacing multiplier applied. Could include that as part of the xmldoc.

Reply:
I get that, but I wanted to confirm that it's not being used for DI or anything. ie. removing it here should have no effect, correct?

Reply:
Yes, that's correct.